projects/packages/my-jetpack/_inc/components/golden-token/index.jsx

```diff
@@ -0,0 +1,167 @@
+import { Button, Col, Container } from '@automattic/jetpack-components';
+import { Modal } from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+import classNames from 'classnames';
+import PropTypes from 'prop-types';
+import React, { useCallback, useRef, useState } from 'react';
+import { JetpackBackupLogo, JetpackLogo, JetpackScanLogo } from './logos';
+import styles from './styles.module.scss';
+
+/**
+ * Component that displays a golden token experience.
+ *
+ * @param {object} props - Component props.
+ * @param {Function} props.redeemClick - Callback function to handle redeem click.
+ * @param {object} props.userConnectionData - Connected user data.
+ * @param {Function} props.onClickGoBack - Callback function to handle go back click.
+ * @param {Array} props.purchases - Array of Object of purchases.
+ * @param {boolean} props.fetchingPurchases - Are purchases being fetched.
+ * @returns {React.Component} - GoldenToken component.
+ */
+function GoldenTokenModal( {
+	redeemClick,
+	userConnectionData,
+	onClickGoBack,
+	fetchingPurchases,
+	purchases,
+} ) {
+	const [ hasAnimated, setIsAnimating ] = useState( false );
+	const videoRef = useRef( null );
+
+	// Any purchase with the partner_slug of 'goldenticket' is considered a golden token.
+	const goldenToken = purchases.filter( golden => golden.partner_slug === 'goldenticket' );
+	const hasGoldenToken = goldenToken.length > 0;
+
+	const redeemClickHandler = useCallback(
+		e => {
+			redeemClick?.( e );
+			setIsAnimating( true );
+			videoRef.current.play();
+		},
+		[ videoRef, redeemClick ]
+	);
+
+	const maybeReanimate = useCallback( () => {
+		hasAnimated && videoRef.current.play();
+	}, [ hasAnimated ] );
+
+	const modalClassName = classNames( styles.modal, {
+		[ styles.animating ]: hasAnimated,
+	} );
+
+	// They might not have a display name set in wpcom, so fall back to wpcom login or local username.
+	const wpcomUserName =
+		userConnectionData?.currentUser?.wpcomUser?.display_name ||
+		userConnectionData?.currentUser?.wpcomUser?.login ||
+		userConnectionData?.currentUser?.username;
+
+	return (
+		<Modal className={ modalClassName } onRequestClose={ onClickGoBack } isDismissible={ false }>
```

Comment:
I don't see the "Back" button in my tests, I think because of the `isDismissible={ false }`.

If we don't need to dismiss button, maybe we can get rid of the Tracks event (although a Tracks event when the modal is opened may be useful?).

Reply:
The modal is still dismissible by clicking outside of the modal or pressing esc. Will update the naming of things